### PR TITLE
Add Verifying Contracts guide

### PR DESCRIPTION
**Pull request by romanbarbo**
## Introduced changes

This Pull Request adds the Verifying Contracts guide for the `verify` command.

## Checklist

- [x] Updated relevant documentation
- [x] Performed self-review of the code

### CODE REVIEW

File: docs/src/starknet/verify.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    --contract-name SimpleBalance \
+    --verifier walnut \
+    --network mainnet
+
+You are about to submit the entire workspace's code to the third-party chosen verifier at walnut, and the code will be publicly available through walnut's APIs. Are you sure? (Y/n) Y
+
+command: verify
+message: Contract successfully verified
+```
+
```
Comment (Arcticae) [inline docs/src/starknet/verify.md:37]: I think we should add an explanation what happens next in the flow (where to look for the status)

File: docs/src/starknet/verify.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Verifying Contracts
+
+## Overview
+
+Starknet Foundry `sncast` supports verifying Cairo contract classes with the `sncast verify` command by submitting the source code to a selected verification provider.
```
Comment (Arcticae) [inline docs/src/starknet/verify.md:5]: This doesn't explain why you should verify and what it really means

File: docs/src/starknet/verify.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+For detailed CLI description, see [verify command reference](../appendix/sncast/verify.md).
+
+> ⚠️ **Warning**
+> Please be aware that submitting the source code means it will be publicly exposed through the provider's APIs.
```
Comment (Arcticae) [inline docs/src/starknet/verify.md:10]: I think it's also worth explaining here what happens to the code on the backend (how it can be used)

**Review by Arcticae — CHANGES_REQUESTED** [review]

**Comment by barabanovro** [comment]
Thank you for the review. I have addressed the comments

File: docs/src/starknet/verify.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    --verifier walnut \
+    --network mainnet
+
+You are about to submit the entire workspace's code to the third-party chosen verifier at walnut, and the code will be publicly available through walnut's APIs. Are you sure? (Y/n) Y
+
+command: verify
+message: Contract successfully verified
+```
+
+After verification with Walnut verifier, the contract can be debugged using Walnut's transaction step-by-step debugger.
```
Comment (Arcticae) [inline docs/src/starknet/verify.md:40]: If this is a general section (see the title marking that you did at `Example`), this does not belong here 

Also i don't think this is relevant at all

Comment (Arcticae) [reply docs/src/starknet/verify.md:40]: It still lacks the explanation where should i look for the confirmation, the "green check mark" that i'm after when doing the verification etc.

Is there a URL that i can use?

**Review by Arcticae — CHANGES_REQUESTED** [review]

**Review by Arcticae — APPROVED** [review]

**Pull request merged**